tools/functional-tester/etcd-tester/stresser.go

```diff
@@ -99,7 +100,7 @@ func (s *stresser) run(ctx context.Context, kvc pb.KVClient) {
 	defer s.wg.Done()
 
 	for {
-		if err := s.rateLimiter.Wait(ctx); err == context.Canceled {
+		if err := s.rateLimiter.Wait(ctx); err == context.Canceled || ctx.Err() == context.Canceled {
```

Comment:
> possible for the timer to expire and ctx.Err() evaluates to nil

Right, this check is racey. I will find a correct way to detect the cancellation.

Thanks.